### ISSUE TITLE
Refactor and expand data types.

### ISSUE THREAD

**Issue by joseedil** [reanimate/reanimate-svg#15]
This is being done in the refactor-datatypes branch, but it breaks the API.

The rational is the following:

- [Attributes.hs](https://github.com/reanimate/reanimate-svg/blob/refactor-datatypes/src/Graphics/SvgTree/Types/Attributes.hs) will hold datatypes for all SVG [attributes and attributes groups](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute).
- ContentTypes.hs will hold datatypes for all SVG [content types](https://developer.mozilla.org/en-US/docs/Web/SVG/Content_type). So far those are in [CssTypes.hs](https://github.com/reanimate/reanimate-svg/blob/refactor-datatypes/src/Graphics/SvgTree/CssTypes.hs).
- [Elements.hs](https://github.com/reanimate/reanimate-svg/blob/refactor-datatypes/src/Graphics/SvgTree/Types/Elements.hs) will contain the [elements](https://developer.mozilla.org/en-US/docs/Web/SVG/Element) data types.

**Comment by joseedil** [comment]
Content types:

- [x] \<angle\>
- [x] \<anything\>
- [ ] \<clock-value\>
- [x] \<color\>
- [x] \<coordinate\>
- [x] \<frequency\>
- [ ] \<funcIRI\>
- [ ] \<ICCColor\>
- [x] \<integer\>
- [ ] \<IRI\>
- [x] \<Length\>
- [x] \<length-percentage\> - This is commented so far in favor of \<length\> which includes a percentage format.
- [x] \<list-of-Ts\>
- [x] \<name\>
- [x] \<number\> - This one breaks the parser and almost everything else. Earlier implementation confounded \<number\> and \<length\>.
- [x] \<number-optional-number\>
- [x] \<opacity-value\>
- [x] \<paint\>
- [x] \<percentage\>
- [x] \<time\>
- [x] \<transform-list\>
- [x] \<units\> - This is not part of the standard but it is useful to define some attributes.
- [x] \<URL\>

**Comment by joseedil** [comment]
Attibutes: this is a [big list](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute), here are just what is missing, many of them because of the content types above.

- [ ] begin
- [ ] by
- [ ] clip-path
- [ ] contentScriptType
- [ ] contentStyleType
- [ ] cursor
- [ ] d - Very relevant for paths!
- [ ] decelerate - Definition is missing in the SVG website.
- [ ] display
- [ ] dur
- [ ] end
- [ ] filter
- [ ] font-family
- [ ] font-variant
- [ ] from
- [ ] g1 
- [ ] g2 
- [ ] glyph-name
- [ ] gradientTransform
- [ ] hreflang - Definition is missing in the SVG website.
- [ ] id
- [ ] keyPoints
- [ ] keySplines
- [ ] keyTimes
- [ ] 
- [ ] 
- [ ] 
- [ ]

**Comment by joseedil** [comment]
Those are the SVG [elements](https://developer.mozilla.org/en-US/docs/Web/SVG/Element). Everything is missing so far.

- [ ] \<a\>
- [ ] \<animate\>
- [ ] \<animateMotion\>
- [ ] \<animateTransform\>
- [x] \<circle\>
- [ ] \<clipPath\>
- [ ] \<color-profile\>
- [ ] \<defs\>
- [ ] \<desc\>
- [ ] \<discard\>
- [ ] \<ellipse\>
- [ ] \<feBlend\>
- [ ] \<feColorMatrix\>
- [ ] \<feComponentTransfer\>
- [ ] \<feComposite\>
- [ ] \<feConvolveMatrix\>
- [ ] \<feDiffuseLighting\>
- [ ] \<feDisplacementMap\>
- [ ] \<feDistantLight\>
- [ ] \<feDropShadow\>
- [ ] \<feFlood\>
- [ ] \<feFuncA\>
- [ ] \<feFuncB\>
- [ ] \<feFuncG\>
- [ ] \<feFuncR\>
- [ ] \<feGaussianBlur\>
- [ ] \<feImage\>
- [ ] \<feMerge\>
- [ ] \<feMergeNode\>
- [ ] \<feMorphology\>
- [ ] \<feOffset\>
- [ ] \<fePointLight\>
- [ ] \<feSpecularLighting\>
- [ ] \<feSpotLight\>
- [ ] \<feTile\>
- [ ] \<feTurbulence\>
- [ ] \<filter\>
- [ ] \<foreignObject\>
- [ ] \<g\>
- [ ] \<hatch\>
- [ ] \<hatchpath\>
- [ ] \<image\>
- [ ] \<line\>
- [ ] \<linearGradient\>
- [ ] \<marker\>
- [ ] \<mask\>
- [ ] \<mesh\>
- [ ] \<meshgradient\>
- [ ] \<meshpatch\>
- [ ] \<meshrow\>
- [ ] \<metadata\>
- [ ] \<mpath\>
- [ ] \<path\>
- [ ] \<pattern\>
- [ ] \<polygon\>
- [ ] \<polyline\>
- [ ] \<radialGradient\>
- [ ] \<rect\>
- [ ] \<script\>
- [ ] \<set\>
- [ ] \<solidcolor\>
- [ ] \<stop\>
- [ ] \<style\>
- [ ] \<svg\>
- [ ] \<switch\>
- [ ] \<symbol\>
- [ ] \<text\>
- [ ] \<textPath\>
- [ ] \<title\>
- [ ] \<tspan\>
- [ ] \<unknown\>
- [ ] \<use\>
- [ ] \<view\\>

**Comment by Lemmih** [comment]
How much has to be implemented before we can merge this into master?
